internal/app/handlers/dynamic/stubbing-provider.go

```diff
@@ -0,0 +1,108 @@
+package dynamic
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/machacekondra/fakeovirt/internal/app/router"
+
+	"github.com/gorilla/pat"
+	"github.com/machacekondra/fakeovirt/api/stubbing"
+)
+
+//
+type StubbingProvider struct {
+	handlerConfigurators map[string]HandlersConfigurator
+	router               *router.ReplacableDelegatingRouter
+}
+
+// HandlersConfigurator creates initial router configuration
+type HandlersConfigurator func(*pat.Router)
+
+// NewStubbingHandler creates new stubbing handler
+func NewStubbingHandler(router *router.ReplacableDelegatingRouter) *StubbingProvider {
+	handler := StubbingProvider{
+		router: router,
+	}
+	return &handler
+}
+
+// Configure adds paths to handlers serving stubbing requests and ones made by the handlersConfigurators to the router held by the StubbingProvider
+func (h *StubbingProvider) Configure(handlersConfigurators map[string]HandlersConfigurator) {
+	h.handlerConfigurators = handlersConfigurators
+	h.configure()
+	h.configureProvidedHandlers()
+}
+
+func (h *StubbingProvider) configureProvidedHandlers() {
+	for key := range h.handlerConfigurators {
+		handlersConfigurator := h.handlerConfigurators[key]
+		handlersConfigurator(h.router.Delegate())
+	}
+}
+
+func (h *StubbingProvider) configure() {
+	delegate := h.router.Delegate()
+	delegate.Post("/stub", h.Stub())
+	delegate.Post("/reset", h.Reset())
+}
+
+// Reset restores original paths in the router held by the StubbingProvider
+func (h *StubbingProvider) Reset() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		newRouter := h.purge()
+
+		h.configure()
```

Comment:
This looks similar to logic in `Configure`. Is it possible to reuse it?

Reply:
Taking into account the differences between `Configure` and `Reset` (especially conditional branching in the latter) I don't think that there is a way to extract more common elements.